src/measure_theory/group/action.lean

```diff
@@ -0,0 +1,114 @@
+some basic properties of such measures.
+-/
+
+open_locale ennreal nnreal pointwise topological_space
+open measure_theory measure_theory.measure set function
+
+namespace measure_theory
+
+variables {G M α : Type*} [measurable_space α]
+
+/-- Measure `μ : measure α` is invariant under an additive action of `M` on `α` if for any
+measurable set `s : set α` and `c : M`, the measure of its preimage under `λ x, c +ᵥ x` is equal to
+the measure of `s`. -/
+class vadd_invariant_measure (M α : Type*) [has_vadd M α] [measurable_space α]
+  (μ : measure α) : Prop :=
+(measure_preimage_vadd [] : ∀ (c : M) ⦃s : set α⦄, measurable_set s → μ ((λ x, c +ᵥ x) ⁻¹' s) = μ s)
+
+/-- Measure `μ : measure α` is invariant under a multiplicative action of `M` on `α` if for any
+measurable set `s : set α` and `c : M`, the measure of its preimage under `λ x, c • x` is equal to
+the measure of `s`. -/
+@[to_additive] class smul_invariant_measure (M α : Type*) [has_scalar M α] [measurable_space α]
+  (μ : measure α) : Prop :=
+(measure_preimage_smul [] : ∀ (c : M) ⦃s : set α⦄, measurable_set s → μ ((λ x, c • x) ⁻¹' s) = μ s)
+
+namespace smul_invariant_measure
+
+@[to_additive] instance zero [has_scalar M α] : smul_invariant_measure M α 0 := ⟨λ _ _ _, rfl⟩
+
+@[to_additive] instance add [has_scalar M α] {μ ν : measure α} [smul_invariant_measure M α μ]
+  [smul_invariant_measure M α ν] : smul_invariant_measure M α (μ + ν) :=
+⟨λ c s hs, show _ + _ = _ + _,
+  from congr_arg2 (+) (measure_preimage_smul μ c hs) (measure_preimage_smul ν c hs)⟩
+
+@[to_additive] instance smul [has_scalar M α] {μ : measure α} [smul_invariant_measure M α μ]
+  (c : ℝ≥0∞) : smul_invariant_measure M α (c • μ) :=
+⟨λ a s hs, show c • _ = c • _, from congr_arg ((•) c) (measure_preimage_smul μ a hs)⟩
+
+@[to_additive] instance smul_nnreal [has_scalar M α] {μ : measure α} [smul_invariant_measure M α μ]
+  (c : ℝ≥0) : smul_invariant_measure M α (c • μ) :=
+smul_invariant_measure.smul c
+
+end smul_invariant_measure
+
+variables [group G] [mul_action G α] [measurable_space G] [has_measurable_smul G α]
+
+@[simp, to_additive]
+lemma map_smul (c : G) (μ : measure α . volume_tac) [smul_invariant_measure G α μ] :
+  map ((•) c) μ = μ :=
+ext $ λ s hs, (map_apply (measurable_const_smul c) hs).trans
+  (smul_invariant_measure.measure_preimage_smul _ _ hs)
+
+@[to_additive]
+lemma measure_preserving_smul (c : G) (μ : measure α . volume_tac) [smul_invariant_measure G α μ] :
+  measure_preserving ((•) c) μ μ :=
+⟨measurable_const_smul c, map_smul c μ⟩
+
+@[simp, to_additive] lemma measure_preimage_smul (c : G) (μ : measure α . volume_tac)
+  [smul_invariant_measure G α μ] (s : set α) : μ ((•) c ⁻¹' s) = μ s :=
+(measure_preserving_smul c μ).measure_preimage_emb (measurable_embedding_const_smul c) s
+
+@[simp, to_additive] lemma measure_smul_set (c : G) (μ : measure α . volume_tac)
+  [smul_invariant_measure G α μ] (s : set α) : μ (c • s) = μ s :=
+by rw [← preimage_smul_inv, measure_preimage_smul]
+
```

Comment:
can you open a section `is_minimal` here?

Reply:
Done.